<robot name="six_dof_arm"  xmlns:xacro="http://www.ros.org/wiki/xacro">

	<xacro:include filename="$(find urdf_tutorials)/urdf/6dof_advanced/links_joints.xacro" />
  <xacro:include filename="$(find urdf_tutorials)/urdf/6dof_advanced/name.xacro" />



	<ros_box name="${link_00_name}"
              origin_rpy="0 0 0" origin_xyz="0 0 0"
              mass="359.99"
              ixx="32.7" ixy="-7.3275E-15" ixz="-7.4508E-18"
              iyy="32.7" iyz="2.9965E-16"
              izz="60"
              size="0.5 0.5 0.15" />

    <ros_joint name="joint_one" type="revolute"
           axis_xyz="0 0 1"
           origin_rpy="0 0 0" origin_xyz="0 0 0.075"
           parent="${link_00_name}" child="link_one"
           effort_limit="200" lower_limit="-3.14" upper_limit="3.14" velocity_limit="0.2" />

    <ros_3D_mesh name="link_one"
              origin_rpy="0 0 0" origin_xyz="0 0 0"
              mass="4.6167"
              ixx="0.069637" ixy="-1.8435E-05" ixz="-1.1246E-05"
              iyy="0.069576" iyz="-0.0059016"
              izz="0.017056"
              meshfile="package://urdf_tutorials/meshes/6dof/link_one.STL"
              meshscale="1 1 1" />

    <ros_joint name="joint_two" type="revolute"
           axis_xyz="0 1 0"
           origin_rpy="0 0 0" origin_xyz="0 0 0.3048"
           parent="link_one" child="link_two"
           effort_limit="200" lower_limit="-3.14" upper_limit="3.14" velocity_limit="0.2" />

    <ros_3D_mesh name="link_two"
              origin_rpy="0 0 0" origin_xyz="0 0 0"
              mass="8.6679"
              ixx="0.15166" ixy="3.6507E-08" ixz="-2.2357E-07"
              iyy="0.15385" iyz="1.0954E-07"
              izz="0.019839"
              meshfile="package://urdf_tutorials/meshes/6dof/link_two.STL"
              meshscale="1 1 1" />


    <ros_joint name="joint_three" type="revolute"
           axis_xyz="0 1 0"
           origin_rpy="0 0 0" origin_xyz="0 0 0.3302"
           parent="link_two" child="link_three"
           effort_limit="200" lower_limit="-3.14" upper_limit="3.14" velocity_limit="0.2" />

    <ros_3D_mesh name="link_three"
              origin_rpy="0 0 0" origin_xyz="0 0 0"
              mass="4.7843"
              ixx="0.030521" ixy="1.0714E-09" ixz="-7.9633E-10"
              iyy="0.031396" iyz="0.005677"
              izz="0.010855"
              meshfile="package://urdf_tutorials/meshes/6dof/link_three.STL"
              meshscale="1 1 1" />

    <ros_joint name="joint_three_two" type="revolute"
           axis_xyz="0 0 1"
           origin_rpy="0 0 0" origin_xyz="0 0 0.1905"
           parent="link_three" child="link_three_two"
           effort_limit="200" lower_limit="-3.14" upper_limit="3.14" velocity_limit="0.2" />

    <ros_3D_mesh name="link_three_two"
              origin_rpy="0 0 0" origin_xyz="0 0 0"
              mass="4.7843"
              ixx="0.030521" ixy="1.0714E-09" ixz="-7.9633E-10"
              iyy="0.031396" iyz="0.005677"
              izz="0.010855"
              meshfile="package://urdf_tutorials/meshes/6dof/link_three_two.STL"
              meshscale="1 1 1" />

    <ros_joint name="joint_four" type="revolute"
           axis_xyz="0 1 0"
           origin_rpy="0 0 0" origin_xyz="0 0 0.1905"
           parent="link_three_two" child="link_four"
           effort_limit="200" lower_limit="-3.14" upper_limit="3.14" velocity_limit="0.2" />

    <ros_3D_mesh name="link_four"
              origin_rpy="0 0 0" origin_xyz="0 0 0"
              mass="3.23"
              ixx="0.011615" ixy="3.0049E-09" ixz="-2.3516E-09"
              iyy="0.013505" iyz="0.0016812"
              izz="0.0069931"
              meshfile="package://urdf_tutorials/meshes/6dof/link_four.STL"
              meshscale="1 1 1" />

    <ros_joint name="joint_five" type="revolute"
           axis_xyz="0 0 1"
           origin_rpy="0 0 0" origin_xyz="0 0 0.1143"
           parent="link_four" child="link_five"
           effort_limit="200" lower_limit="-3.14" upper_limit="3.14" velocity_limit="0.2" />

    <ros_3D_mesh name="link_five"
              origin_rpy="0 0 0" origin_xyz="0 0 0"
              mass="0.96751"
              ixx="0.0012933" ixy="-1.7602E-11" ixz="-1.1137E-18"
              iyy="0.0013496" iyz="-1.4677E-18"
              izz="0.0024173"
              meshfile="package://urdf_tutorials/meshes/6dof/link_five.STL"
              meshscale="1 1 1" />

    <ros_joint name="gripper_one_joint" type="prismatic"
           axis_xyz="1 0 0"
           origin_rpy="0 0 0" origin_xyz="-0.06 0 0.0881"
           parent="link_five" child="gripper_one"
           effort_limit="200" lower_limit="0" upper_limit="0.03" velocity_limit="0.2" />

    <ros_box name="gripper_one"
              origin_rpy="0 0 0" origin_xyz="0 0 0"
              mass="0.2"
              ixx="0.0395416666667" ixy="0" ixz="0"
              iyy="0.106208333333" iyz="0"
              izz="0.106208333333"
              size="0.01 0.05 0.1" />

    <ros_joint name="gripper_two_joint" type="prismatic"
           axis_xyz="-1 0 0"
           origin_rpy="0 0 0" origin_xyz="0.06 0 0.0881"
           parent="link_five" child="gripper_two"
           effort_limit="200" lower_limit="0" upper_limit="0.03" velocity_limit="0.2" />

    <ros_box name="gripper_two"
              origin_rpy="0 0 0" origin_xyz="0 0 0"
              mass="0.2"
              ixx="0.0395416666667" ixy="0" ixz="0"
              iyy="0.106208333333" iyz="0"
              izz="0.106208333333"
              size="0.01 0.05 0.1" />


             



              













           






	<gazebo>
  		<plugin name="gazebo_ros_control" filename="libgazebo_ros_control.so"> 
  			<robotNamespace>/six_dof_arm</robotNamespace>   
  		</plugin>
	</gazebo> 
</robot>
